src/csharp/Grpc.Core/Internal/BatchContextSafeHandle.cs

```diff
@@ -92,5 +101,22 @@ protected override bool ReleaseHandle()
             Native.grpcsharp_batch_context_destroy(handle);
             return true;
         }
+
+        void IOpCompletionCallback.OnComplete(bool success)
+        {
+            try
+            {
+                CompletionCallback(success, this);
+            }
+            catch (Exception e)
+            {
+                Logger.Error(e, "Exception occured while invoking batch completion delegate.");
+            }
+            finally
+            {
+                CompletionCallback = null;
```

Comment:
A little hung up here, why does `CompletionCallback` need to be nullled? Same question for `RequestCallContextSafeHandle`

Reply:
If we decide to pool BatchContextSafeHandle objects and reuse them. it's going to be important to release references to the callback so that they don't stay referenced forever and can be cleaned up by the GC.